src/pybind/mgr/volumes/fs/nfs.py

```diff
@@ -708,3 +709,48 @@ def list_nfs_cluster(self):
         except Exception as e:
             log.exception("Failed to list NFS Cluster")
             return getattr(e, 'errno', -1), "", str(e)
+
+    def _show_nfs_cluster_info(self, cluster_id):
+        self._set_cluster_id(cluster_id)
+        completion = self.mgr.list_daemons(daemon_type='nfs')
+        self.mgr._orchestrator_wait([completion])
+        orchestrator.raise_if_exception(completion)
+        host_ip = {}
+        # Here completion.result is a list DaemonDescription objects
+        for cluster in completion.result:
+            if self.cluster_id == cluster.service_id():
+                """
+                getaddrinfo sample output: [(<AddressFamily.AF_INET: 2>,
+                <SocketKind.SOCK_STREAM: 1>, 6, 'xyz', ('172.217.166.98',2049)),
+                (<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '',
+                ('2404:6800:4009:80d::200e', 2049, 0, 0))]
+                """
+                try:
+                    host_ip[cluster.daemon_id] = {
+                            "hostname": cluster.hostname,
+                            "IP": list(set([ip[4][0] for ip in socket.getaddrinfo(
+                                cluster.hostname, 2049, flags=socket.AI_CANONNAME,
+                                type=socket.SOCK_STREAM)])),
+                            "Port": 2049  # Default ganesha port
+                            }
+                except socket.gaierror:
+                    continue
```

Comment:
maybe still report the daemon exists, but set the IP to a NoneType?

Reply:
I am not sure if we should set it.